src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -236,6 +241,20 @@ private void ValidateProcessedBlock(Block suggestedBlock, ProcessingOptions opti
         ReceiptsTracer.StartNewBlockTrace(block);
 
         _beaconBlockRootHandler.ApplyContractStateChanges(block, spec, _stateProvider);
+
+        if (spec.IsEip2935Enabled)
+        {
+            // TODO: find a better way to handle this - no need to have this check everytime
+            //      this would just be true on the fork block
+            BlockHeader parentHeader = _blockTree.FindParentHeader(block.Header, BlockTreeLookupOptions.None);
+            if (parentHeader is not null && parentHeader!.Timestamp < spec.Eip2935TransitionTimestamp)
+                _blockhashStore.InitHistoryOnForkBlock(block.Header);
+            else
+                _blockhashStore.AddParentBlockHashToState(block.Header);
+
+        }
```

Comment:
All this logic could be moved into the `Store` as `ApplyHistoryBlockHashes`. Store already has `blockFinder` and could just decide what to do.

This way we could remove `blocktree` from constructor and only have this being dependant on the store.